### ISSUE TITLE
requests with compound filters cause errors

### ISSUE THREAD

**Issue by ripienaar** [choria-io/go-choria#160]
```
ERRO[467967] Could not create Request to validate Secure Request with: The JSON body from the SecureRequest is not a valid Request message: Supplied JSON document is not a valid Request message: envelope.filter.compound.0: Invalid type. Expected: string, given: array
ERRO[467967] Could not cache Client Certificate: The JSON body from the SecureRequest is not a valid Request message: Supplied JSON document is not a valid Request message: envelope.filter.compound.0: Invalid type. Expected: string, given: array
ERRO[467967] Could not decode incoming request: SecureRequest message created from the Transport Message did not pass security validation  component=server identity=dev1-choria.devco.net
```

There's some JSON Schema bug in the protocol I suspect but did not investigate closely.

I am inclined to just not support compound filters at all though in choria

**Comment by ripienaar** [comment]
This is for sure wrong:

https://github.com/choria-io/go-protocol/blob/a6dae85b97fdee0fba68033d920aa3f4d5ffd9b3/protocol/v1/schema/request.json#L84

```json
"compound": {
                            "type":"array",
                            "description": "List of compound filters to be applied in an AND fashion",
                            "items": {
                                "type":"string"
                            }
                        },
```

also wrong here

https://github.com/choria-io/go-protocol/blob/a6dae85b97fdee0fba68033d920aa3f4d5ffd9b3/protocol/filter.go#L21

really have to decide if i want to support these things long term

**Comment by ripienaar** [comment]
It's actually a array of hash like this:

https://github.com/puppetlabs/marionette-collective/blob/c2d10c64af26bf6b24a5f0aee5b4caf31a61300f/lib/mcollective/matcher/parser.rb#L117

meh

**Comment by ripienaar** [comment]
Fixed in go-protocol 1.3.0

Need to address this now:

https://github.com/choria-io/go-choria/blob/71d264f121c91c2f9bd151d630ea3bfbfb104f5f/server/discovery/discovery.go#L82-L85

**Comment by ripienaar** [comment]
an actual compound filter:

```
 "compound"=> [
      [
        {
          "fstatement"=> {
            "r_compare"=>"30",
            "operator"=>">",
            "value"=>"total_time",
            "name"=>"resource",
            "params"=>nil
          }
        }
     ],
     [
        {
          "statement"=>"environment=development"
        },
        {
          "or"=>"or"
        },
        {
          "statement"=>"customer=acme"}
        ]
    ],
```
